server/src/main/java/org/candlepin/model/CPRestrictions.java

```diff
@@ -24,21 +27,25 @@
 import java.util.Iterator;
 import java.util.List;
 
-
+import javax.inject.Inject;
 
 /**
  * The CPRestrictions class provides utility Criterion building methods to be used with Hibernate's
  * fluent-style query building.
  */
 public class CPRestrictions {
+
+    private Configuration config;
+
     private static class CPLikeExpression extends LikeExpression {
         public CPLikeExpression(String property, String value, Character escape, boolean ignoreCase) {
             super(property, value, escape, ignoreCase);
         }
     }
 
-    protected CPRestrictions() {
-        throw new UnsupportedOperationException("CPRestriction should not be instantiated");
+    @Inject
+    public CPRestrictions(Configuration config) {
+        this.config = config;
```

Comment:
Is it possible to keep this a static factory? Personally, I would prefer to pass the block size in as a parameter and keep the CPRestrictions factory object consistent with Hibernate's Restrictions factory. Also simplifies some constructors and avoids some extra mocking and setup in the classes that use them.